### PR TITLE
migrate: tf_image_dashboard

### PR DESCRIPTION
**Pull request by wchargin**
Initial migration by @manivaradarajan; closing fixes by @wchargin.

Test Plan:
With `TB_POLYMER3=1 bazel run //tensorboard`, verified that:

  - images render properly;
  - the global “actual image size” toggle works;
  - individual “actual image size” toggles (clicking on images) work;
  - the brightness and contrast adjustments work;
  - the brightness and contrast “Reset” buttons work;
  - manually selecting/deselecting runs works;
  - entering a run regex filter works;
  - entering a tag regex filter works;
  - changing the pagination limit in settings works;
  - the “no data” screen renders correctly.

Also verified functionality after test sync: <http://cl/325324290>.

Co-authored-by: Mani Varadarajan <maniv@google.com>

### CODE REVIEW

File: tensorboard/plugins/image/polymer3/tf_image_dashboard/tf-image-loader.ts [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+        mainImageContainer.textContent = '';
+        (PolymerDom.dom(mainImageContainer) as any).appendChild(img);
+        this.set('_isImageLoading', false);
+      })
+      .bind(this);
+    img.style.filter = `contrast(${contrastPercentage}%) `;
+    img.style.filter += `brightness(${brightnessAdjustment})`;
+    // Load the new image.
+    this.set('_isImageLoading', true);
+    img.src = (currentStep as any).url;
```
Comment (stephanwlee) [inline tensorboard/plugins/image/polymer3/tf_image_dashboard/tf-image-loader.ts:355]: Is tsetse okay with `img.src`?

Comment (wchargin) [reply tensorboard/plugins/image/polymer3/tf_image_dashboard/tf-image-loader.ts:355]: I guess it is—the sync (<http://cl/325324290>) builds and works. (That
is a bit interesting, though.)

File: tensorboard/plugins/image/polymer3/tf_image_dashboard/tf-image-dashboard.ts [hunk 2]
```diff
@@ -59,8 +48,8 @@ <h3 class="tooltip-container">Brightness adjustment</h3>
               <paper-slider
                 min="0"
                 max="2"
-                snaps
-                pin
+                snaps=""
```
[the remark this answers is not in the document]
Comment (wchargin) [reply tensorboard/plugins/image/polymer3/tf_image_dashboard/tf-image-dashboard.ts]: Good to know; thanks! Reverted these and some others (`git grep '=""'`).
Still seems to work, including spinners.

File: tensorboard/plugins/image/polymer3/tf_image_dashboard/tf-image-loader.ts [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+        </div>
+      </template>
+      <template is="dom-if" if="[[_hasMultipleSteps]]">
+        <div>
+          <paper-slider
+            id="steps"
+            immediate-value="{{_stepIndex}}"
+            max="[[_maxStepIndex]]"
+            max-markers="[[_maxStepIndex]]"
+            snaps=""
```
[the remark this answers is not in the document]
Comment (wchargin) [reply tensorboard/plugins/image/polymer3/tf_image_dashboard/tf-image-loader.ts]: Done.

**Pull request merged**